### PR TITLE
Add definition of done ADR

### LINKED ISSUE #629: Need to add definition of done to documentation
**Is your feature request related to a problem? Please describe.**
We are not documenting the definition of done.

**Describe the solution you'd like**
Add this agreed DoD as an ADR, summarise it and ensure it's in our documentation:

- An accessibility swarm can take time to organise. If we block merging into Psammead for time to arrange this, it could slow down development.
- We want to test components as early as possible, before it’s been merged and then integrated into the application.
- We want to ensure we have a balance that keeps high standards yet allows fast development.

***When working on PRs that are not yet ready for production use***

 - Use an alpha version: `package.json` / `package-lock.json` version has format `0.0.0.alpha.0` & alpha tag.
 - This is an industry standard way to enable publishing a package that is not yet ready for use in a production environment.
 - We should never use `alpha` versions of components in our production application.

 - Follow the [developers checklist](https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-developers) to help you code with accessibility in mind
- Each Psammead PR that changes the UI should include checking local code with assistive technology (which could just be testing with VoiceOver on your Mac)
- No change with how Testers test these PRs

***Changing from alpha to production-ready***

- Create an issue for changing from an alpha version to a production version (e.g. `1.0.0.alpha.0` to `1.1.0`).
- The relevant PR to update the version can only be merged after an [Accessibility Swarm](https://bbc-news.github.io/accessibility-news-and-you/accessibility-swarms) has taken place. 
- We use our resources and time effectively and therefore have different levels of swarm - Developer and Team. As a general rule you can follow the following guidance, these are not hard and fast rules, confirm what makes sense for your project with the Accessibility Lead:

- Developer accessibility swarm for:
    - Components that are intended to be used by other components and not intended to be used in isolation i.e. a component part or sub component of a feature
    - For example, [image caption](https://bbc.github.io/psammead/?path=/story/caption--default) would be considered a component part or sub component
    - Or a small change to a live component e.g. changing a colour etc.

- Team accessibility swam for:
    - Something that won’t form part of another component e.g. a full component or feature
    - For example, [figure with caption](https://bbc.github.io/psammead/?path=/story/figure--containing-image-imageplaceholder-copyright-and-caption)
    - Or a substantial change to a live component

- If the accessibility swarm is being carried out on an open PR, this should be done after code reviews, when in the A11y review column, prior to being moved to ready for test
- Accessibility swarms (developer or team) should always complete the [swarm checklist](https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-testers) including testing with all [supported AT](https://bbc-news.github.io/accessibility-news-and-you/accessibility-and-supported-assistive-technology).
- If there are any P1 (priority 1) accessibility issues, they must be addressed before moving from an alpha version to a production version.
- Full browser testing on supported browsers can be carried out at this time. https://github.com/bbc/psammead#bar_chart-support-levels

**Describe alternatives you've considered**
n/a we really need this defined

**Testing notes**
[Tester to complete]

Dev insight: n/a just documentation changes

**Additional context**

### PR DESCRIPTION
**Pull request by Bopchy**
Resolves #629 

**Overall change:** _Add definition of done ADR._

**Code changes:**

- _Add definition of done ADR to documentation/_
---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+### Decision Drivers
+
+* An accessibility swarm can take time to organise. If we block merging into Psammead for time to arrange this, it could slow down development.
+* We want to test components as early as possible - before they've been merged and integrated into the application.
+* We want to ensure we have a balance that keeps high standards yet allows fast development.
+
+## When working on PRs that are not yet ready for production use:
+
+Use an alpha version - is where `package.json` / `package-lock.json` version has format `0.0.0.alpha.0` & `alpha tag`.
```
Comment (12) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]: Should `tag` be in backticks?

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+- This is an industry standard way to enable publishing a package that is not yet ready for use in a production environment.
+- We should never use alpha versions of components in our production application.
+
+Follow the [developers checklist](https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-developers) to help you code with accessibility in mind
+
+- Each Psammead PR that changes the UI should include checking local code with assistive technology (which could just be testing with VoiceOver on your Mac)
+
+No change with how Testers test these PRs.
```
Comment (12) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
No change with how test engineers test these PRs.
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+Create an issue for changing from an alpha version to a production version (e.g. `1.0.0.alpha.0` to `1.1.0`)
+
+- The relevant PR to update the version can only be merged after an [Accessibility Swarm](https://bbc-news.github.io/accessibility-news-and-you/accessibility-swarms) has taken place.
+
+We use our resources and time effectively and therefore have different levels of swarms - Developer and Team. As a general rule, you can follow the following guidance (these are not hard and fast rules. Confirm what makes sense for your project with the Accessibility Lead):
+
+#### Developer accessibility swarm for:
+- Components that are intended to be used by other components and not intended to be used in isolation i.e. a component part or sub component of a feature
+
+For example, [image caption](https://bbc.github.io/psammead/?path=/story/caption--default) would be considered a component part or sub component
```
Comment (12) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]: Should this be nested in the list item above?

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+Follow the [developers checklist](https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-developers) to help you code with accessibility in mind
+
+- Each Psammead PR that changes the UI should include checking local code with assistive technology (which could just be testing with VoiceOver on your Mac)
+
+No change with how Testers test these PRs.
+
+## When changing from alpha to production-ready:
+
+Create an issue for changing from an alpha version to a production version (e.g. `1.0.0.alpha.0` to `1.1.0`)
```
Comment (DenisHdz) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
Create an issue for changing from an alpha version to a production version (e.g. `1.0.0.alpha.0` to `1.1.0`).
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+* We want to ensure we have a balance that keeps high standards yet allows fast development.
+
+## When working on PRs that are not yet ready for production use:
+
+Use an alpha version - is where `package.json` / `package-lock.json` version has format `0.0.0.alpha.0` & `alpha tag`.
+
+- This is an industry standard way to enable publishing a package that is not yet ready for use in a production environment.
+- We should never use alpha versions of components in our production application.
+
+Follow the [developers checklist](https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-developers) to help you code with accessibility in mind
```
Comment (DenisHdz) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
Follow the [developers checklist](https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-developers) to help you code with accessibility in mind.
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+### Decision Drivers
+
+* An accessibility swarm can take time to organise. If we block merging into Psammead for time to arrange this, it could slow down development.
+* We want to test components as early as possible - before they've been merged and integrated into the application.
+* We want to ensure we have a balance that keeps high standards yet allows fast development.
+
+## When working on PRs that are not yet ready for production use:
+
+Use an alpha version - is where `package.json` / `package-lock.json` version has format `0.0.0.alpha.0` & `alpha tag`.
+
```
Comment (DenisHdz) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md:28]: I would also mention that we should add the following text:
```
# ⛔️ This is an alpha component ⛔️

This component is currently tagged as alpha and is not suitable for production use. Following the passing of an accessibility review this component will be marked as ready for production and the alpha tag removed.
```
at the top of the README.

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md (hunk 3 above)
Comment (Bopchy) [reply documentation/architecture-decision-records/ADR0003-definition_of_done.md]: Made it a nested block quote to associate it with the list item above it

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md (hunk 1 above)
Comment (Bopchy) [reply documentation/architecture-decision-records/ADR0003-definition_of_done.md]: Changed it to `alpha` tag.

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+### Decision Drivers
+
+* An accessibility swarm can take time to organise. If we block merging into Psammead for time to arrange this, it could slow down development.
+* We want to test components as early as possible - before they've been merged and integrated into the application.
+* We want to ensure we have a balance that keeps high standards yet allows fast development.
+
+## When working on PRs that are not yet ready for production use:
+
+Use an alpha version - is where `package.json` / `package-lock.json` version has format `0.0.0.alpha.0` & `alpha` tag, and add the below text to the top of the README:
```
Comment (j-pendlebury) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
Use an alpha version, which is done by changing the version in `package.json` / `package-lock.json` to `0.0.0.alpha.0`, containing an `alpha` tag. 

Text below should also be added to the top of the README:
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+  > ### ⛔️ This is an alpha component ⛔️
+  > This component is currently tagged as alpha and is not suitable for production use. Following the passing of an accessibility review this component will be marked as ready for production and the alpha tag removed.
+
+- This is an industry standard way to enable publishing a package that is not yet ready for use in a production environment.
+- We should never use alpha versions of components in our production application.
+
+Follow the [developers checklist](https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-developers) to help you code with accessibility in mind.
+
+- Each Psammead PR that changes the UI should include checking local code with assistive technology (which could just be testing with VoiceOver on your Mac)
```
Comment (j-pendlebury) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
- Each Psammead PR that changes the UI should include checking local code with assistive technology (which could just be testing with VoiceOver on your Mac).
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+No change with how test engineers test these PRs.
+
+## When changing from alpha to production-ready:
+
+Create an issue for changing from an alpha version to a production version (e.g. `1.0.0.alpha.0` to `1.1.0`).
```
Comment (j-pendlebury) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
Create an issue for updating component's alpha version to a production version (e.g. `1.0.0.alpha.0` to `1.1.0`).
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+- The relevant PR to update the version can only be merged after an [Accessibility Swarm](https://bbc-news.github.io/accessibility-news-and-you/accessibility-swarms) has taken place.
+
+We use our resources and time effectively and therefore have different levels of swarms - Developer and Team. As a general rule, you can follow the following guidance (these are not hard and fast rules. Confirm what makes sense for your project with the Accessibility Lead):
```
Comment (j-pendlebury) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
We use our resources and time effectively and therefore have different levels of swarms - Developer and Team. As a general rule, you can follow the following guidance below, however these are not hard and fast rules. You should confirm what makes sense for your project with the Accessibility Lead.
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+#### Developer accessibility swarm for:
+- Components that are intended to be used by other components and not intended to be used in isolation i.e. a component part or sub component of a feature
+
+  > For example, [image caption](https://bbc.github.io/psammead/?path=/story/caption--default) would be considered a component part or sub component
+
+- A small change to a live component
+
+  > For example - changing a color, etc.
```
Comment (j-pendlebury) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
  > For example - changing a colour, etc.
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+#### Team accessibility swarm for:
+- Something that won’t form part of another component - e.g. a full component or feature
```
Comment (j-pendlebury) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
- Something that won’t form part of another component, e.g. a full component or feature
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+  > An example of this is [figure with caption](https://bbc.github.io/psammead/?path=/story/figure--containing-image-imageplaceholder-copyright-and-caption)
+
+- A substantial change to a live component
+
+If the accessibility swarm is being carried out on an open PR, this should be done after code reviews, when in the A11y review column, prior to being moved to ready for test
```
Comment (j-pendlebury) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
If the accessibility swarm is being carried out on an open PR, this should be done after code reviews, when in the A11y review column, prior to being moved to ready for test.
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+
+Accessibility swarms (developer or team) should always complete the [swarm checklist](https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-testers) including testing with all [supported AT](https://bbc-news.github.io/accessibility-news-and-you/accessibility-and-supported-assistive-technology).
```
Comment (j-pendlebury) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
During accessibility swarms (developer or team), you should always complete the [swarm checklist](https://bbc-news.github.io/accessibility-news-and-you/accessibility-news-and-testers) including testing with all [supported AT](https://bbc-news.github.io/accessibility-news-and-you/accessibility-and-supported-assistive-technology).
```

File: documentation/architecture-decision-records/ADR0003-definition_of_done.md [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+
+If there are any P1 (priority 1) accessibility issues, they must be addressed before moving from an alpha version to a production version.
+
+Full browser testing on supported browsers can be carried out at this time. https://github.com/bbc/psammead#bar_chart-support-levels
```
Comment (j-pendlebury) [inline documentation/architecture-decision-records/ADR0003-definition_of_done.md]:
```suggestion
Full browser testing on [supported browsers](https://github.com/bbc/psammead#bar_chart-support-levels) can be carried out at this time.
```

**Review by j-pendlebury — CHANGES_REQUESTED** [review]
Just some suggestions from me 🙈

**Review by j-pendlebury — APPROVED** [review]

**Review by DenisHdz — APPROVED** [review]

**Pull request merged**